src/context.ts

```diff
@@ -27,8 +30,20 @@ function getUserId(token?: string): string {
       throw new ApolloError(err.name === 'TokenExpiredError' ? 'Token expired' : 'Invalid authentication token');
     }
     return decoded;
-  }) as unknown as { userId: string };
-  return verificationTokenResult.userId;
+  }) as unknown as { sessionId: string };
+  const { sessionId } = verificationTokenResult;
+  const session = await xprisma.session.findUniqueOrThrow({
+    where: {
+      id: sessionId,
+    },
+    include: {
+      creator: true,
+    },
+  });
+  if (session.revokedAt && session.revokedAt < new Date()) {
+    throw new ApolloError('Session expired');
+  }
```

Comment:
I believe this whole part should be part of the session model, eg `prisma.session.getUser(token)`

Reply:
Also, I don't think you need to check `session.revokedAt < new Date()` since there is no functionality to revoke session in the future that you provide